docs/migration-v8.md

```diff
@@ -1 +1,515 @@
+// callbacks, there is no need to define it twice.
+ResiliencePipeline pipeline = new ResiliencePipelineBuilder()
+    .AddConcurrencyLimiter(permitLimit: 100, queueLimit: 50)
+    .Build();
+
+// Create a generic pipeline with concurrency limiter. Because ResiliencePipeline<T> supports both sync and async
+// callbacks, there is no need to define it twice.
+ResiliencePipeline<HttpResponseMessage> pipelineT = new ResiliencePipelineBuilder<HttpResponseMessage>()
+    .AddConcurrencyLimiter(permitLimit: 100, queueLimit: 50)
+    .Build();
+```
+<!-- endSnippet -->
+
+## Migrating `Polly.Context`
+
+`Polly.Context` has been succeeded by `ResilienceContext`. Here are the main changes:
+
+- `ResilienceContext` is pooled for enhanced performance and isn't directly creatable. Instead, use the `ResilienceContextPool` to get an instance.
+- Directly attaching custom data is supported by `Context`, whereas `ResilienceContext` employs the `ResilienceContext.Properties` property.
+- Both `PolicyKey` and `PolicyWrapKey` are no longer a part of `ResilienceContext`. They've been relocated to `ResiliencePipelineBuilder` and are now used for [telemetry](advanced/telemetry.md#metrics).
+- The `CorrelationId` property has been removed. For similar functionality, you can either use `System.Diagnostics.Activity.Current.Id` or attach your custom id using `ResilienceContext.Properties`.
+- Additionally, `ResilienceContext` introduces the `CancellationToken` property.
+
+### `Context` in v7
+
+<!-- snippet: migration-context-v7 -->
+```cs
+// Create context
+Context context = new Context();
+
+// Create context with operation key
+context = new Context("my-operation-key");
+
+// Attach custom properties
+context["prop-1"] = "value-1";
+context["prop-2"] = 100;
+
+// Retrieve custom properties
+string value1 = (string)context["prop-1"];
+int value2 = (int)context["prop-2"];
+```
+<!-- endSnippet -->
+
+### `ResilienceContext` in v8
+
+<!-- snippet: migration-context-v8 -->
+```cs
+// Create context
+ResilienceContext context = ResilienceContextPool.Shared.Get();
+
+// Create context with operation key
+context = ResilienceContextPool.Shared.Get("my-operation-key");
+
+// Attach custom properties
+context.Properties.Set(new ResiliencePropertyKey<string>("prop-1"), "value-1");
+context.Properties.Set(new ResiliencePropertyKey<int>("prop-2"), 100);
+
+// Retrieve custom properties
+string value1 = context.Properties.GetValue(new ResiliencePropertyKey<string>("prop-1"), "default");
+int value2 = context.Properties.GetValue(new ResiliencePropertyKey<int>("prop-2"), 0);
+
+// Return the context to the pool
+ResilienceContextPool.Shared.Return(context);
+```
+<!-- endSnippet -->
+
+For more details, refer to the [Resilience context](advanced/resilience-context.md) documentation.
+
+## Migrating no-op policy
+
+- For `Policy.NoOp` or `Policy.NoOpAsync`, now use `ResiliencePipeline.Empty`.
```

Comment:
nit: I would probably use the same `now use` or `switch to` for both since they are there just twice and I actually reread the first line again to make sure what I was supposed to do.